### PR TITLE
Updated WebRequests with a WebResponse Callback

### PR DESCRIPTION
**Pull request by austinv900**
Added a new descriptive class callback
Added HEAD as a Method option

### CODE REVIEW

**Comment by Visagalis** [comment]
that in callback and response `V2` looks totally out of pattern.

**Comment by austinv900** [comment]
@Visagalis hmm?

File: Oxide.Core/Libraries/WebRequests.cs [hunk 1]
```diff
@@ -100,6 +134,17 @@ public WebRequest(string url, Action<int, string> callback, Plugin owner)
...
 
+            /// <summary>
+            /// Initializes a new instance of the WebRequest class
+            /// </summary>
+            /// <param name="url"></param>
+            /// <param name="owner"></param>
+            /// <param name="callback"></param>
+            public WebRequest(string url, Plugin owner, Action<WebResponse> callback) : this(url, null, owner)
+            {
+                CallbackV2 = callback;
```
Comment (Visagalis) [inline Oxide.Core/Libraries/WebRequests.cs]: name `CallbackV2` looks very out of pattern.

File: Oxide.Core/Libraries/WebRequests.cs [hunk 2]
```diff
@@ -395,6 +636,25 @@ public void Enqueue(string url, string body, Action<int, string> callback, Plugi
...
+        /// Enqueues a DELETE, GET, PATCH, POST, HEAD, or PUT web request
+        /// </summary>
+        /// <param name="url"></param>
+        /// <param name="body"></param>
+        /// <param name="callback"></param>
+        /// <param name="owner"></param>
+        /// <param name="method"></param>
+        /// <param name="headers"></param>
+        /// <param name="timeout"></param>
+        [LibraryFunction("EnqueueV2")]
```
Comment (Visagalis) [inline Oxide.Core/Libraries/WebRequests.cs]: `EnqueueV2` looks out of pattern. Probably need to use another naming for new methods/parameters before deprecating others.

File: Oxide.Core/Libraries/WebRequests.cs [hunk 3]
```diff
@@ -41,6 +65,11 @@ public class WebRequest
             /// </summary>
             public Action<int, string> Callback { get; }
 
+            /// <summary>
+            /// Gets the callback delegate
+            /// </summary>
+            public Action<WebResponse> CallbackV2 { get; }
```
Comment (Visagalis) [inline Oxide.Core/Libraries/WebRequests.cs]: Same issue as above.

File: Oxide.Core/Libraries/WebRequests.cs (hunk 1 above)
Comment (austinv900) [reply Oxide.Core/Libraries/WebRequests.cs]: Can't really call it the same thing.

File: Oxide.Core/Libraries/WebRequests.cs (hunk 3 above)
Comment (austinv900) [reply Oxide.Core/Libraries/WebRequests.cs]: Can't really call it the same thing.

File: Oxide.Core/Libraries/WebRequests.cs (hunk 2 above)
Comment (austinv900) [reply Oxide.Core/Libraries/WebRequests.cs]: Library functions don't support same name. These changes are temporary and will be changed later most likely with the release of uMod. Right now this is just a band-aid so we arn't breaking every plugin that uses a webrequest.

**Comment by lukespragg** [comment]
Manually merged with changes, thanks!

**Pull request closed without merging**